virttest/utils_test/qemu/__init__.py

```diff
@@ -665,19 +665,22 @@ def unplug_memory(self, vm, name):
         """
         devices = []
         qid_dimm = "dimm-%s" % name
+        qid_virtio_mem = "virtio_mem-%s" % name
         qid_mem = "mem-%s" % name
+
         try:
             dimm = vm.devices.get_by_qid(qid_dimm)[0]
         except IndexError:
             LOG.warn("'%s' is not used by any dimm" % qid_mem)
         else:
-            step = "Unplug pc-dimm '%s'" % qid_dimm
-            error_context.context(step, LOG.info)
-            _, ver_out = vm.devices.simple_unplug(dimm, vm.monitor)
-            if ver_out is False:
-                raise exceptions.TestFail("Verify unplug memory failed")
-            devices.append(dimm)
-            self.update_vm_after_unplug(vm, dimm)
+            self.device_unplug(dimm, qid_dimm, devices, vm)
+
+        try:
+            vmem = vm.devices.get_by_qid(qid_virtio_mem)[0]
+        except IndexError:
+            LOG.warn("'%s' is not used by any virtio-mem" % qid_mem)
+        else:
+            self.device_unplug(vmem, qid_virtio_mem, devices, vm)
```

Comment:
I think dimm and virtio-mem can in the one `try` code block and use `for` loop, but currently we only 2 types of devices, so not an urgent thing, up to you.

Reply:
Updated, thanks !